Tests/UnitTests/Core/TrackFinding/CombinatorialKalmanFilterTests.cpp

```diff
@@ -22,327 +20,181 @@
-  CKFSourceLinkSelector::Config sourcelinkSelectorConfig = {
-      // global default valies
-      {makeId(), {8.0, 10}},
-      // pixel layer 2 chi2/nTestSourceLinks cutoff: 8.0/5
-      {makeId(2, 2), {8.0, 5}},
-      // pixel layer 4 chi2/nTestSourceLinks cutoff: 7.0/5
-      {makeId(2, 4), {7.0, 5}},
-      // pixel volume chi2/nTestSourceLinks cutoff: 7.0/5
-      {makeId(2), {7.0, 5}},
-      // strip volume chi2/nTestSourceLinks cutoff: 8.0/5
-      {makeId(3), {8.0, 5}},
-  };
-  CombinatorialKalmanFilter cKF(rPropagator);
-
-  // Run the CombinaltorialKamanFitter for track finding from different starting
-  // parameter
-  for (const auto& [trackId, pos] : startingPos) {
-    // Set initial parameters for the particle track
-    Covariance cov;
-    cov << pow(10_um, 2), 0., 0., 0., 0., 0., 0., pow(10_um, 2), 0., 0., 0., 0.,
-        0., 0., pow(0.0002, 2), 0., 0., 0., 0., 0., 0., pow(0.0002, 2), 0., 0.,
-        0., 0., 0., 0., 0.0001, 0., 0., 0., 0., 0., 0., 1.;
-    Vector3D rPos =
-        pos + Vector3D{0, 10_um * gauss(generator), 10_um * gauss(generator)};
-    double rPhi = 0_degree + 0.0002 * gauss(generator);
-    double rTheta = 90_degree + 0.0002 * gauss(generator);
-    CurvilinearTrackParameters rStart(makeVector4(rPos, 42_ns), rPhi, rTheta,
-                                      1_GeV, 1_e, cov);
-
-    const Surface* rSurface = &rStart.referenceSurface();
-
-    auto logger =
-        getDefaultLogger("CombinatorialKalmanFilter", Logging::VERBOSE);
-    CombinatorialKalmanFilterOptions<TestSourceLinkCalibrator,
-                                     CKFSourceLinkSelector>
-        ckfOptions(tgContext, mfContext, calContext, TestSourceLinkCalibrator(),
-                   CKFSourceLinkSelector(sourcelinkSelectorConfig),
-                   LoggerWrapper{*logger}, PropagatorPlainOptions(), rSurface);
-
-    // Find the track(s)
-    auto combKalmanFilterRes = cKF.findTracks(sourcelinks, rStart, ckfOptions);
-    BOOST_CHECK(combKalmanFilterRes.ok());
-
-    auto foundTrack = *combKalmanFilterRes;
-    auto& fittedStates = foundTrack.fittedStates;
-    auto& trackTips = foundTrack.trackTips;
-
-    for (const auto& tip : trackTips) {
-      std::vector<size_t> trackIds;
-      fittedStates.visitBackwards(tip, [&](const auto& trackState) {
-        trackIds.push_back(trackState.uncalibrated().sourceId);
-      });
-
-      BOOST_CHECK_EQUAL(trackIds.size(), 6);
-
-      size_t numFakeHit = 0;
-      for (const auto& id : trackIds) {
-        numFakeHit = numFakeHit + (id != trackId ? 1 : 0);
-      }
-
-      // Check if there are fake hits from other tracks
-      BOOST_CHECK_EQUAL(numFakeHit, 0);
+    for (auto& sl : measurements.sourceLinks) {
+      sourceLinks.emplace_back(std::move(sl));
     }
+  }
 
-    // Construct a starting parameter near the tracker exit
-    Vector3D rPosOuter(-1. * rPos.x(), rPos.y(), rPos.z());
-    CurvilinearTrackParameters rStartOuter(makeVector4(rPosOuter, 42_ns), rPhi,
```

Comment:
Do you have the test for the outward->inward mode somewhere?

Reply:
I exists for the Kalman fitter. Was there also one for the CKF that I missed?

Reply:
Now, I see it. I will re-add the missing outward-inward test.

Reply:
Thanks!

Reply:
As discussed privately, the test is added but disabled for now. Understanding and fixing it will be left to a separate PR.